kiele

```diff
@@ -141,6 +147,47 @@ run_upload_report() {
     rm report.zip
 }
 
+run_compile() {
+    run_pyiele compile "$@"
+
+}
+
+run_testrunner() {
+    local run_file
+    if [[ $# -gt 0 ]] && [ "$1" == "--help" ]; then
+      run_pyiele test --help
+      exit
+    fi
+    if ! nc -z "$kiele_host" "$kiele_port"; then
+        fatal "No listener found on port $kiele_port"
+    fi
+    if [[ $# -gt 0 ]]; then
+      if [[ -f "$1" ]] || [[ -d "$1" ]]; then
+        run_file="$1"; shift
+        if [[ "$run_file" == *.py ]]; then
+            python3 "$run_file"
+        else
+            run_pyiele test --file "$run_file" --port "$kiele_port" "$@"
+        fi
+      else
+        run_pyiele test --port "$kiele_port" "$@"
+      fi
+    else
+      run_pyiele test --port "$kiele_port" "$@"
+    fi
+}
+
+run_blackbox() {
+    if [[ $# -gt 0 ]] && [ "$1" == "--help" ]; then
+      run_pyiele test --help
+      exit
+    fi
+    if ! nc -z "$kiele_host" "$kiele_port"; then
+        fatal "No listener found on port $kiele_port"
+    fi
```

Comment:
Factor out a `check_listener` function in the Utilities section at the top, and use that.

Reply:
Done.